### PR TITLE
CustomMetrics in HPA MetricsClient

### PR DESCRIPTION
**Pull request by mwielgus**
Depends on #20155

### CODE REVIEW

**Comment by fgrzadkowski** [comment]
Apart from few minor comments this LGTM. I only reviewed commit "CustomMetrics in HPA MetricsClient"

File: pkg/controller/podautoscaler/metrics/metrics_client.go [hunk 1]
```diff
@@ -46,16 +45,19 @@ type MetricsClient interface {
 	// (e.g. 70 means that an average pod uses 70% of the requested CPU)
 	// and the time of generation of the oldest of utilization reports for pods.
 	GetCPUUtilization(namespace string, selector map[string]string) (*int, time.Time, error)
+
+	// GetCustomMetric return the average value of the given custom custom metrics from the
```
Comment (fgrzadkowski) [inline pkg/controller/podautoscaler/metrics/metrics_client.go]: s/return/returns
s/custom custom/custom

File: pkg/controller/podautoscaler/metrics/metrics_client.go [hunk 2]
```diff
@@ -64,98 +66,111 @@ type metricDefinition struct {
...
+	if missing || requestSum == 0 {
+		return 0, 0, time.Time{}, fmt.Errorf("some pods do not have request for cpu")
 	}
-	glog.Infof("Sum of %s requested: %v", resourceName, sum)
-	avg := resource.MustParse(fmt.Sprintf("%dm", sum.MilliValue()/int64(len(podList.Items))))
-	request = &avg
-	consumption, timestamp, err = h.getForPods(resourceName, namespace, podNames)
+	glog.Infof("Sum of CPU requested: %d", requestSum)
+	requestAvg := requestSum / int64(len(podList.Items))
+	// consumption is already averaged and in millis.
```
Comment (fgrzadkowski) [inline pkg/controller/podautoscaler/metrics/metrics_client.go]: Start with capital letter.

File: pkg/controller/podautoscaler/metrics/metrics_client.go (hunk 1 above)
Comment (mwielgus) [inline pkg/controller/podautoscaler/metrics/metrics_client.go]: Done.

File: pkg/controller/podautoscaler/metrics/metrics_client.go (hunk 2 above)
Comment (mwielgus) [inline pkg/controller/podautoscaler/metrics/metrics_client.go]: Done

**Comment by mwielgus** [comment]
Adding LGTM label per @fgrzadkowski comment.

**Pull request merged**